dll/cpl/input/settings_page.c

```diff
@@ -394,18 +563,28 @@ BOOL EnableProcessPrivileges(LPCWSTR lpPrivilegeName, BOOL bEnable)
     return Ret;
 }
 
-static VOID
+static INT_PTR
 OnNotifySettingsPage(HWND hwndDlg, LPARAM lParam)
 {
     LPNMHDR header = (LPNMHDR)lParam;
 
     switch (header->code)
     {
-        case LVN_ITEMCHANGED:
+        case TVN_SELCHANGED:
         {
-            if (header->idFrom == IDC_KEYLAYOUT_LIST)
+            SetControlsState(hwndDlg);
+        }
+        break;
+
+        case TVN_ITEMEXPANDING:
+        {
+            // FIXME: Prevent collapse (COMCTL32 is buggy)
+            // https://bugs.winehq.org/show_bug.cgi?id=53727
```

Comment:
Question is whether they have it fixed in their current code or not (since we don't use the latest synced comctl32)

Reply:
Not fixed yet in the latest WineHQ.
```c
    if (bUser)
	TREEVIEW_SendExpanding(infoPtr, item, action);
```